src/lib.rs

```diff
@@ -269,6 +270,23 @@ impl ConfiguredOidc {
         return None;
     }
 
+    /// Get the host of the HTTP request
+    /// The host is searched in the request headers. If the host is found, the value is returned.
+    fn get_host(&self) -> Option<String> {
+
+        let host = if self.get_http_request_header(":authority").is_some() {
```

Comment:
You could do this with a chain of [`or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.or_else) calls. 
```rust
let host = self.get_http_request_header(":authority")
    .or_else(|| self.get_http_request_header("X-Forwarded-Host"))
    .or_else(|| self.get_http_request_header("host"));
```